### DISCUSSION TITLE
Sending a POST request to Next.js API route is returning a 500 Internal Server Error

### DISCUSSION THREAD

**Discussion by Armaghan15** [vercel/next.js#40657] [Help]
![image](https://user-images.githubusercontent.com/69668009/190914256-1dbbdb1d-940b-499a-9159-f194cdbe005f.png)

![image](https://user-images.githubusercontent.com/69668009/190914268-86ba4fdc-379c-42d9-b204-4515f03b12f9.png)

I am sending a POST request to my next.js api route, but it returns a 500 internal server error. Code shown above.

**Comment by icyJoseph** [comment]
Got a repo to look at?

I'd include the `parse` inside the try catch, because `parse` is an operation that can fail anyway.

Then inside the catch, note that converting errors to JSON is not as easy as you have it. In your case it'll be serialised to `{}`, instead log to the terminals what's going on, that is `console.error(error)`, and inspect it there.

We want to figure out if our operations are crashing the API or if it happens in the outer code that handles the API. Most likely it is one of our operations, perhaps we have invalid JSON as `req.body`, but we need to investigate. Again, got a repo to look at?
